### PR TITLE
more dht log optimizations

### PR DESCRIPTION
**Pull request by aldenml**

### CODE REVIEW

File: src/kademlia/find_data.cpp [hunk 1]
```diff
@@ -110,10 +109,12 @@ void find_data::start()
 void find_data::got_write_token(node_id const& n, std::string write_token)
 {
 #ifndef TORRENT_DISABLE_LOGGING
-	get_node().observer()->log(dht_logger::traversal
-		, "[%p] adding write token '%s' under id '%s'"
-		, static_cast<void*>(this), aux::to_hex(write_token).c_str()
-		, aux::to_hex(n).c_str());
+	auto logger = get_node().observer();
+	if (logger != nullptr && logger->should_log(dht_logger::traversal))
```
Comment (arvidn) [inline src/kademlia/find_data.cpp]: I think it would be clearer to add the { and } here, since the statement is multiple lines

Comment (aldenml) [inline src/kademlia/find_data.cpp]: ok, I will add in the others too

File: src/kademlia/get_peers.cpp [hunk 2]
```diff
@@ -72,10 +72,12 @@ void get_peers_observer::reply(msg const& m)
 			char const* end = peers + n.list_at(0).string_length();
 
 #ifndef TORRENT_DISABLE_LOGGING
+			auto logger = get_observer();
 			bdecode_node const id = r.dict_find_string("id");
```
Comment (arvidn) [inline src/kademlia/get_peers.cpp]: technically, this lookup could be moved inside the ``should_log()`` check as well

File: src/kademlia/get_peers.cpp [hunk 3]
```diff
@@ -94,18 +96,20 @@ void get_peers_observer::reply(msg const& m)
 			// assume it's uTorrent/libtorrent format
 			read_endpoint_list<tcp::endpoint>(n, peer_list);
 #ifndef TORRENT_DISABLE_LOGGING
+			auto logger = get_observer();
 			bdecode_node const id = r.dict_find_string("id");
```
Comment (arvidn) [inline src/kademlia/get_peers.cpp]: and here I think

**Pull request merged**